components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -322,45 +333,59 @@ else if (ConfigurationFacade.getInstance().isDumbMode() && authenticatedIdPs.isE
+                            && isAuthenticationFlowHandler) {
+                        sendToMultiOptionPage(stepConfig, request, context, response, authenticatorNames);
+                    }
                     return;
                 } else {
                     // else send to the multi option page.
-                    if (log.isDebugEnabled()) {
-                        log.debug("Sending to the Multi Option page");
-                    }
-                    diagnosticLog.info("Sending to the Multi Option page");
-                    Map<String, String> parameterMap = getAuthenticatorConfig().getParameterMap();
-                    String showAuthFailureReason = null;
-                    if (parameterMap != null) {
-                        showAuthFailureReason = parameterMap.get(FrameworkConstants.SHOW_AUTHFAILURE_RESON_CONFIG);
-                        if (log.isDebugEnabled()) {
-                            log.debug("showAuthFailureReason has been set as : " + showAuthFailureReason);
-                        }
-                        diagnosticLog.info("showAuthFailureReason has been set as : " + showAuthFailureReason);
-                    }
-                    String retryParam = "";
-
-                    if (stepConfig.isRetrying()) {
-                        context.setCurrentAuthenticator(null);
-                        retryParam = "&authFailure=true&authFailureMsg=login.fail.message";
-                    }
-
-                    try {
-                        request.setAttribute(FrameworkConstants.RequestParams.FLOW_STATUS, AuthenticatorFlowStatus
-                                .INCOMPLETE);
-                        response.sendRedirect(getRedirectUrl(request, response, context, authenticatorNames,
-                                showAuthFailureReason, retryParam, loginPage));
-                    } catch (IOException e) {
-                        diagnosticLog.error("Server error occurred. Error message: " + e.getMessage());
-                        throw new FrameworkException(e.getMessage(), e);
-                    }
-
+                    sendToMultiOptionPage(stepConfig, request, context, response, authenticatorNames);
                     return;
                 }
             }
         }
     }
 
+    private void sendToMultiOptionPage(StepConfig stepConfig, HttpServletRequest request, AuthenticationContext context,
+                                       HttpServletResponse response, String authenticatorNames)
+            throws FrameworkException {
+
+        String loginPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL();
+        if (log.isDebugEnabled()) {
+            log.debug("Sending to the Multi Option page");
+        }
+        diagnosticLog.info("Sending to the Multi Option page");
+        Map<String, String> parameterMap = getAuthenticatorConfig().getParameterMap();
+        String showAuthFailureReason = null;
+        if (parameterMap != null) {
```

Comment:
Shall we use `MapUtils`?

Reply:
Fixed https://github.com/wso2/carbon-identity-framework/pull/3533/commits/e2e1a48558bfa645abdcdeec06bba479bf62a72e